### PR TITLE
Lighteater is now recursive

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

The light eater will now recursively break lights inside of things.

## Why It's Good For The Game

Someone with a flashlight in their bag will no longer be immune to nightmares/

## Testing Photographs and Procedure

image

## Changelog
:cl:
fix: Nightmare's arm blade will now destroy lights inside backpacks
/:cl:

### CODE REVIEW

**Comment by FriendlyContractor** [comment]
Suggestion - make it also work on an item person is pulling. Its practically impossible to click on tiny light source being pulled by someone while they are moving and very cheesy way to fight nightmare. Also does this work on hardhats being worn by cyborgs ?

**Comment by PowerfulBacon** [comment]
> Suggestion - make it also work on an item person is pulling. Its practically impossible to click on tiny light source being pulled by someone while they are moving and very cheesy way to fight nightmare. Also does this work on hardhats being worn by cyborgs ?

It does

**Comment by AnCopper** [comment]
Small thing, but can you fix the thing where the light eater does a whack sound instead of a cutting sound? It's a blade.

**Comment by dannydonk** [comment]
Does this fix helmet lights and gunlights not being broken by lighteater?

**Comment by PowerfulBacon** [comment]
> Does this fix helmet lights and gunlights not being broken by lighteater?

Yes, anything inside anything thats on you will be destroyed if it has a light.

File: code/game/atoms.dm [hunk 1]
```diff
@@ -803,7 +803,8 @@
   * Called when lighteater is called on this.
   */
 /atom/proc/lighteater_act(obj/item/light_eater/light_eater)
-	return
+	for(var/atom/A in src)
```
Comment (Archanial) [inline code/game/atoms.dm:808]: Check contents

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/game/atoms.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/atoms.dm:808]: Isn't this the same thing?

**Comment by PowerfulBacon** [comment]
uhhhhhhhhh thats a lot of signals actually
Perhaps this should just have component lights register a signal to their holder

**Comment by ItsTisme** [comment]
Does this work with my light eater PR? may wana check that. 



> uhhhhhhhhh thats a lot of signals actually Perhaps this should just have component lights register a signal to their holder

Technicaly added a signal something worth using maby?

**Comment by PowerfulBacon** [comment]
This is a terrible version of https://github.com/BeeStation/BeeStation-Hornet/pull/7014

**Pull request closed without merging**